### PR TITLE
Move aggregate_reports to separate updater

### PR DESCRIPTION
**Pull request by antooni**
Resolves L2B-1771

### CODE REVIEW

File: packages/backend/src/test/mockReports.ts [hunk 1]
```diff
@@ -53,39 +103,35 @@ export const BALANCES: BalanceRecord[] = [
...
-    ],
-  },
+const FUTURE_BALANCES = BALANCES.map((balance) => ({
+  ...balance,
+  timestamp: NOW.add(1, 'hours'),
+}))
+
+const REPORTS = createReports(PRICES, BALANCES, PROJECTS)
+const FUTURE_REPORTS = createReports(FUTURE_PRICES, FUTURE_BALANCES, PROJECTS)
+// We need this mock for now, it ensures that the OP token is included in the
```
Comment (mateuszradomski) [inline packages/backend/src/test/mockReports.ts:113]: You'll forget about this comment in a week, mark it `TODO` or something that sticks out

**Review by mateuszradomski — APPROVED** [review]

File: packages/backend/src/core/reports/AggregatedReportUpdater.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      this.projects,
+      timestamp,
+    )
+
+    await this.aggregatedReportStatusRepository.add({
+      configHash: this.configHash,
+      timestamp,
+    })
+
+    await this.aggregatedReportRepository.addOrUpdateMany(aggregatedReports),
```
Comment (michalsidzej) [inline packages/backend/src/core/reports/AggregatedReportUpdater.ts:68]: We should add these 2 in a transaction, or at least, in an opposite order. It's better to sync the same timestamp twice than to display untrue data.

**Review by michalsidzej — CHANGES_REQUESTED** [review]

File: packages/backend/src/core/reports/AggregatedReportUpdater.ts (hunk 2 above)
Comment (antooni) [reply packages/backend/src/core/reports/AggregatedReportUpdater.ts:68]: fully agree, nice catch, will update it in other places as well

File: packages/backend/src/core/reports/AggregatedReportUpdater.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    await this.aggregatedReportRepository.addOrUpdateMany(aggregatedReports)
+
+    await this.aggregatedReportStatusRepository.add({
+      configHash: this.configHash,
+      timestamp,
+    })
+
+    this.logger.info('Report updated', { timestamp: timestamp.toNumber() })
+  }
```
Comment (michalsidzej) [inline packages/backend/src/core/reports/AggregatedReportUpdater.ts:71]: Nit pick, but those spaces are not necessary IMO

**Review by michalsidzej — APPROVED** [review]
LGTM

**Pull request merged**